pgmpy/inference/continuous_sampling.py

```diff
@@ -0,0 +1,302 @@
+    Public Methods:
+    ---------------
+    sample()
+    generate_sample()
+
+    Example:
+    --------
+    #
+
+    References
+    -----------
+    Matthew D. Hoffman, Andrew Gelman, The No-U-Turn Sampler: Adaptively
+    Setting Path Lengths in Hamiltonian Monte Carlo. Journal of
+    Machine Learning Research 15 (2014) 1351-1381
+    """
+
+    def __init__(self, model, Lambda, grad_log_pdf,
+                 discretize_time=LeapFrog, delta=0.65):
+
+        BaseHMC.__init__(self, model=model, grad_log_pdf=grad_log_pdf,
+                         discretize_time=discretize_time, delta=delta)
+
+        self.Lambda = Lambda
+
+    def _find_resonable_epsilon(self, theta, epsilon_app=1):
+        """
+        Method for choosing initial value of epsilon
+
+        References
+        -----------
+        Matthew D. Hoffman, Andrew Gelman, The No-U-Turn Sampler: Adaptively
+        Setting Path Lengths in Hamiltonian Monte Carlo. Journal of
+        Machine Learning Research 15 (2014) 1351-1381
+        Algorithm 4 : Heuristic for choosing an initial value of epsilon
+        """
+        # momentum = N(0, I)
+        momentum = np.matrix(np.reshape(np.random.normal(0, 1, len(theta)), (len(theta), 1)))
+
+        # Take a single step in time
+        theta_bar, momentum_bar = self.discretize_time(self.grad_log_pdf, self.model,
+                                                       theta, momentum, epsilon_app).discretize_time()
+        # Parameters to help in evaluating P(theta, momentum)
+        _, logp = self.grad_log_pdf(theta, self.model).get_gradient_log_pdf()
+        _, logp_bar = self.grad_log_pdf(theta_bar, self.model).get_gradient_log_pdf()
+
+        # acceptance_prob = P(theta_bar, momentum_bar)/ P(theta, momentum)
+        potential_change = logp_bar - logp  # Negative change
+        kinetic_change = 0.5 * np.float(np.dot(momentum_bar.T, momentum_bar) - np.dot(momentum.T, momentum))
+
+        acceptance_prob = np.exp(potential_change - kinetic_change)
+
+        # a = 2I[acceptance_prob] -1
+        a = 2 * (acceptance_prob > 0.5) - 1
+
+        condition = (acceptance_prob ** a) > (2 ** (-a))
+
+        while condition:
+            epsilon_app = (2 ** a) * epsilon_app
+
+            theta_bar, momentum_bar = self.discretize_time(self.grad_log_pdf, self.model,
+                                                           theta, momentum, epsilon_app).discretize_time()
+
+            _, logp = self.grad_log_pdf(theta, self.model).get_gradient_log_pdf()
+            _, logp_bar = self.grad_log_pdf(theta_bar, self.model).get_gradient_log_pdf()
+
+            potential_change = logp_bar - logp
+            kinetic_change = 0.5 * np.float(momentum_bar.T * momentum_bar - momentum.T * momentum)
+
+            acceptance_prob = np.exp(potential_change - kinetic_change)
+
+            condition = (acceptance_prob ** a) > (2 ** (-a))
+
+        return epsilon_app
+
+    def sample(self, theta0, num_adapt, num_samples, epsilon=None):
```

Comment:
@khalibartan Since you have used a for loop in this function, I think you should be able to write `generate_sample` using this method.